### PR TITLE
Enable dollar variable pattern

### PR DESCRIPTION
**Pull request by megcrenshaw**
Enables a specific stylelint rule, after merging in https://github.com/code-dot-org/code-dot-org/pull/47624.

Ran into some weird questions ... namely, variables that are defined in .js files, or variables that aren't used anywhere. @mikeharv would love your eyes on the data modules files.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/storage/dataBrowser/data-browser.module.scss [hunk 1]
```diff
@@ -1,19 +1,17 @@
 @import 'color.scss';
 
-$tableWidth: 400px;
```
Comment (megcrenshaw) [inline apps/src/storage/dataBrowser/data-browser.module.scss:3]: I didn't see where some of these variables were being used, so I deleted them all together

File: apps/style/craft/style.scss [hunk 2]
```diff
@@ -14,7 +14,7 @@ $below-header-background: #646464;
 $blockly-header: $very-dark-gray;
 $flyout-background: #4e4e4e;
 $footer-background: #4e4e4e;
-$footer-text–and-arrow: #969292;
+$footer-text-and-arrow: #969292;
```
Comment (megcrenshaw) [inline apps/style/craft/style.scss:17]: This was weird, but the characters here were slightly different

**Comment by megcrenshaw** [comment]
Apps tests passed with `npm run test:unit`

File: apps/style/craft/style.scss (hunk 2 above)
Comment (evilcrash) [reply apps/style/craft/style.scss:17]: this is an en dash (–), not a dash (-). if there's an en dash in at least one variable every variable should be checked.

Comment (megcrenshaw) [reply apps/style/craft/style.scss:17]: Yep, this was the only weird one –– the linter caught this!

**Review by mikeharv — APPROVED** [review]
These changes make a lot of sense and look good to me. I wasn't sure what the right naming conventions were. It's great to know that linting can help with this going forward!

**Pull request merged**